browser/js/calculator.js

```diff
@@ -22,229 +28,247 @@
-          addDot()
-        } else if ( event.target.matches( elementSelectors.PERCENT ) ) {
-          convertPercent()
+          me.setOperator( event.target.dataset.value )
+        } else if ( event.target.matches( Calculator.elementSelectors.CLEAR ) ) {
+          me.clearCalculator()
+        } else if ( event.target.matches( Calculator.elementSelectors.EQUALS ) ) {
+          me.calculate()
+        } else if ( event.target.matches( Calculator.elementSelectors.TOGGLE_SIGN ) ) {
+          me.toggleNegativeValue()
+        } else if ( event.target.matches( Calculator.elementSelectors.DOT ) ) {
+          me.addDot()
+        } else if ( event.target.matches( Calculator.elementSelectors.PERCENT ) ) {
+          me.convertPercent()
         }
       })
+      this.dom.onkeydown = function( event ) {
+        me.isNumber( event.key )
+          ? ( me.fakeClickEffect( event.key, Calculator.cssClasses.GREY_BUTTON_PRESSED ),
+            me.enterDigit( event.key ) )
+          : me.isOperator( event.key )
+          ? ( me.fakeClickEffect( event.key, Calculator.cssClasses.ORANGE_BUTTON_PRESSED ),
+            me.setOperator( event.key ) )
+          : me.isEquals( event.key )
+          ? ( me.fakeClickEffect( Calculator.buttons.EQUALS, Calculator.cssClasses.ORANGE_BUTTON_PRESSED ),
+            me.calculate() )
+          : me.isClear( event.key )
+          ? ( me.fakeClickEffect( Calculator.buttons.CLEAR, Calculator.cssClasses.GREY_BUTTON_PRESSED ),
+            me.clearCalculator() )
+          : me.isDot( event.key )
+          ? ( me.fakeClickEffect( Calculator.buttons.DOT, Calculator.cssClasses.GREY_BUTTON_PRESSED ),
+            me.addDot() )
+          : null
+      }
   }
-  addListeners()
-
-  function setOutput( text ) {
+  Calculator.prototype.setOutput = function ( text ) {
+    //FIXME: ghetto binding
+    var me = this
     if ( text ) {
-      if ( text.length > data.MAX_LENGTH ) {
-        shrinkFontSize()
+      if ( text.length > this.data.MAX_LENGTH ) {
+        me.shrinkFontSize()
       }
-      document.querySelector( elementSelectors.OUTPUT ).innerHTML = text
+      this.dom.querySelector( Calculator.elementSelectors.OUTPUT ).innerHTML = text
     } else {
-      if ( data.current.length > data.MAX_LENGTH ) {
-        shrinkFontSize()
+      if ( this.data.current.length > this.data.MAX_LENGTH ) {
+        me.shrinkFontSize()
       }
-      document.querySelector( elementSelectors.OUTPUT ).innerHTML = data.current
+      this.dom.querySelector( Calculator.elementSelectors.OUTPUT ).innerHTML = this.data.current
     }
   }
-  function setResult( number ) {
-    data.result = number.toString()
-    data.memory = data.result
+  Calculator.prototype.setResult = function ( number ) {
+    this.data.result = number.toString()
+    this.data.memory = this.data.result
   }
-  function clearResult() {
-    data.result = ''
+  Calculator.prototype
+  .clearResult = function () {
```

Comment:
why are these on the next line? `Calculator.prototype.clearResult = function () {` please :)